### DISCUSSION TITLE
is it normal for nvim-cmp (and luasnip) to not be lazy loaded ?

### DISCUSSION THREAD

**Discussion by mirsella** [LazyVim/LazyVim#3966] [General]
hello

using the luasnip extra, just by launching nvim on the homepage all the nvim-cmp plugins and sources are loaded, because luasnip load nvim-cmp (as i understand it).
looking at the starting performance, nvim-cmp and luasnip is big part of the time taken.

This is the screenshot of  the loaded lazyvim plugin
![image](https://github.com/LazyVim/LazyVim/assets/45905567/f88d25c6-d9bd-4f54-8744-1b72efb2d30b)

thanks !

**Comment by dpetka2001** [comment]
I believe this change happened when there was a change in the `keys` that got moved to the `nvim-cmp` spec in this [commit](https://github.com/LazyVim/LazyVim/commit/da7b7738eb84b00c949f48afabebd39266e61d75), thus removing the lazy-loading from Luasnip.

This might have been avoided if the `keys` [here](https://github.com/LazyVim/LazyVim/blob/d6bda2469787c566801fc0d00e6146c645cbb09f/lua/lazyvim/plugins/coding.lua#L105-L124) were moved inside the `nvim-snippets` spec instead, but I'm not 100% sure. 

You might want to wait for an answer from the maintainer if he sees this or ping him yourself for a more appropriate answer.

**Comment by mirsella** [comment]
@folke sorry for the ping (if he sees the ping?)
i think its worth opening issue ? or is it normal behaviour ?

**Reply by folke** [reply]
I don't use luasnip. Feel free to provide a PR.

**Comment by iamxiaojianzheng** [comment]
After reading your question, I found that my lazyvim starts up the same way, and luasnip loads directly on startup.
Then I tried to configure lazy = true and I found that I could avoid this problem.

```lua
{
    "L3MON4D3/LuaSnip",
    lazy = true,
    -- omit
}
```

![image](https://github.com/user-attachments/assets/55fe8613-938e-43b3-9bfc-80d6836658e9)

**Reply by mirsella** [reply]
ohhhh, i opened a PR but with the `event = "InsertEnter",` to lazyload (because that's what nvim-cmp uses).
really cool that just putting `lazy = true` works. it's because nvim-cmp load cmp-luasnip, which itself loads luasnip. thanks for the idea !
